Bnscli /clitests fail on OSX

## Expected Behavior
`go test github.com/iov-one/weave/cmd/bnscli/clitests` should pass
## Actual Behavior
```
--- FAIL: TestAll (3.65s)
    --- FAIL: TestAll/batch.test (0.01s)
        cli_test.go:52: Below is the cript stderr:
            batch.test: line 5: tempfile: command not found
...
```
`tempfile` does not exist on OSX. We can use [`mktemp`](https://stackoverflow.com/questions/18716658/whats-the-difference-between-tempfile-and-mktemp)

Afterwards it fails on my box with:
```
--- FAIL: TestAll (4.26s)
    --- FAIL: TestAll/keyaddr.test (0.02s)
        cli_test.go:52: Below is the cript stderr:
            base64: invalid option -- d
            Usage:	base64 [-hvD] [-b num] [-i in_file] [-o out_file]
              -h, --help     display this message
              -D, --decode   decodes input
              -b, --break    break encoded string into num character lines
              -i, --input    input file (default: "-" for stdin)
              -o, --output   output file (default: "-" for stdout)


        cli_test.go:54: execution failed: exit status 64
    --- FAIL: TestAll/keygen.test (0.16s)
        cli_test.go:71: want: generated private key length: 64
        cli_test.go:72:  got: generated private key length:       64
        cli_test.go:73: unexpected result
    --- FAIL: TestAll/sign_send_tokens.test (0.03s)
        cli_test.go:52: Below is the cript stderr:
            base64: invalid option -- d
            Usage:	base64 [-hvD] [-b num] [-i in_file] [-o out_file]
              -h, --help     display this message
              -D, --decode   decodes input
              -b, --break    break encoded string into num character lines
              -i, --input    input file (default: "-" for stdin)
              -o, --output   output file (default: "-" for stdout)


        cli_test.go:54: execution failed: exit status 64
FAIL
```
## Steps to Reproduce the Problem
`go test github.com/iov-one/weave/cmd/bnscli/clitests`

## Additional info
We may use a container to make it pass on all environments.

--- a/cmd/bnscli/clitests/batch.test
+++ b/cmd/bnscli/clitests/batch.test
@@ -2,7 +2,7 @@
 
 set -e
 
-msgs=`tempfile`
+msgs=`mktemp`
 
 # Combine all messages in a single file to build a batch transaction.
 bnscli send-tokens -src 'seq:foo/src/1' -dst "seq:foo/dst/1" -amount "2 IOV" -memo "sending 2 IOV" >> $msgs

--- a/cmd/bnscli/clitests/batch_proposal.test
+++ b/cmd/bnscli/clitests/batch_proposal.test
@@ -6,7 +6,7 @@ set -e
 # all of them.
 # The same result could be achieved by running bellow commands in a subprocess
 # and stream the combined output into "as-batch" command.
-msgstream=`tempfile`
+msgstream=`mktemp`
 
 bnscli release-escrow -escrow 4 >> $msgstream
 bnscli send-tokens -src 'seq:foo/src/1' -dst "seq:foo/dst/1" -amount "2 IOV" -memo "sending 2 IOV" >> $msgstream

--- a/cmd/bnscli/clitests/keyaddr.test
+++ b/cmd/bnscli/clitests/keyaddr.test
@@ -5,7 +5,7 @@ set -e
 # bnscli keyaddr command requires a file with the private key in order to sign
 # the transaction. To always produce the same output always use the same input
 # data and private key.
-keyfile=`tempfile`
+keyfile=`mktemp`
 echo 00wZcK6QrPNAXy2Z3KyhbQx9s3n0vq/P32Z7nWnONQ0n9ftEBQnfp57Ig6BRC8mpYUw9RBiIgfDF5AKJi0vzyQ== | base64 -d > $keyfile
 
 bnscli keyaddr -key $keyfile

--- a/cmd/bnscli/clitests/keyaddr.test
+++ b/cmd/bnscli/clitests/keyaddr.test
@@ -6,7 +6,7 @@ set -e
 # the transaction. To always produce the same output always use the same input
 # data and private key.
 keyfile=`mktemp`
-echo 00wZcK6QrPNAXy2Z3KyhbQx9s3n0vq/P32Z7nWnONQ0n9ftEBQnfp57Ig6BRC8mpYUw9RBiIgfDF5AKJi0vzyQ== | base64 -d > $keyfile
+echo 00wZcK6QrPNAXy2Z3KyhbQx9s3n0vq/P32Z7nWnONQ0n9ftEBQnfp57Ig6BRC8mpYUw9RBiIgfDF5AKJi0vzyQ== | base64 --decode > $keyfile
 
 bnscli keyaddr -key $keyfile
 

--- a/cmd/bnscli/clitests/sign_send_tokens.test
+++ b/cmd/bnscli/clitests/sign_send_tokens.test
@@ -13,7 +13,7 @@ fi
 # bnscli sign command requires a file with the private key in order to sign the
 # transaction. To always produce the same output always use the same input data
 # and private key.
-keyfile=`tempfile`
+keyfile=`mktemp`
 echo 00wZcK6QrPNAXy2Z3KyhbQx9s3n0vq/P32Z7nWnONQ0n9ftEBQnfp57Ig6BRC8mpYUw9RBiIgfDF5AKJi0vzyQ== | base64 -d > $keyfile
 
 bnscli send-tokens \

--- a/cmd/bnscli/clitests/sign_send_tokens.test
+++ b/cmd/bnscli/clitests/sign_send_tokens.test
@@ -14,7 +14,7 @@ fi
 # transaction. To always produce the same output always use the same input data
 # and private key.
 keyfile=`mktemp`
-echo 00wZcK6QrPNAXy2Z3KyhbQx9s3n0vq/P32Z7nWnONQ0n9ftEBQnfp57Ig6BRC8mpYUw9RBiIgfDF5AKJi0vzyQ== | base64 -d > $keyfile
+echo 00wZcK6QrPNAXy2Z3KyhbQx9s3n0vq/P32Z7nWnONQ0n9ftEBQnfp57Ig6BRC8mpYUw9RBiIgfDF5AKJi0vzyQ== | base64 --decode > $keyfile
 
 bnscli send-tokens \
 		-src "seq:test/bnscli/1" \

--- a/cmd/cleanproto/tests/test.sh
+++ b/cmd/cleanproto/tests/test.sh
@@ -5,7 +5,7 @@ set -e
 cd $(dirname $(dirname "$0"))
 
 make build
-dest=`tempfile`
+dest=`mktemp`
 ./cleanproto < tests/codec.proto.origin > $dest
 result=`diff tests/codec.proto.gold $dest`
 exit $result

